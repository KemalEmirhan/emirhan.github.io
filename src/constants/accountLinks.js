const accountLinks = [
    { url: 'https://github.com/KemalEmirhan', name: 'Github', icon: '/assets/github.svg', color: '#181717' },
    { url: 'https://www.linkedin.com/in/emirhan-kemal-k√∂sem-259569b3/', name: 'LinkedIn', icon: '/assets/linkedin.svg', color: '#0077B5' },
    { url: 'https://medium.com/@emirhankemalkosem', name: 'Medium', icon: '/assets/medium.svg', color: '#12100E' },
    { url: 'https://twitter.com/KsemEmir', name: 'Twitter', icon: '/assets/twitter.svg', color: '#1DA1F2' },
    // { url: '/assets/resume.pdf', name: 'Resume', icon: '/assets/file.svg', color: '#0077B5' },
];

export default accountLinks;
