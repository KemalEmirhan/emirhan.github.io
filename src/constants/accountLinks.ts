import {
  AccountLinksArraySchema,
  type TAccountLinks,
  type TAccountLink,
} from '@/schemas';

const accountLinksData: TAccountLink[] = [
  {
    url: 'https://github.com/KemalEmirhan',
    name: 'Github',
  },
  {
    url: 'https://www.linkedin.com/in/emirhan-kemal-kosem-259569b3/',
    name: 'Linkedin',
  },
  {
    url: 'https://medium.com/@emirhankemalkosem',
    name: 'Medium',
  },
  {
    url: 'https://twitter.com/KsemEmir',
    name: 'Twitter',
  },
  {
    url: 'https://www.instagram.com/emirhankemalkosem/',
    name: 'Instagram',
  },
];

// Validate and parse account links with Zod
const accountLinks: TAccountLinks =
  AccountLinksArraySchema.parse(accountLinksData);

export default accountLinks;
