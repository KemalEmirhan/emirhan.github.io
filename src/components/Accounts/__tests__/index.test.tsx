import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import Account from '../index';
import accountLinks from '@/constants/accountLinks';

describe('Account Component', () => {
  test('should render', () => {
    const { container } = render(<Account />);
    expect(container).toMatchSnapshot();
  });

  test('should render all account links', () => {
    render(<Account />);

    // Test with actual account links data
    accountLinks.forEach(account => {
      expect(
        screen.getByRole('link', { name: new RegExp(account.name, 'i') })
      ).toBeInTheDocument();
    });
  });

  test('should render account links with correct href attributes', () => {
    render(<Account />);

    // Test with actual account links data
    accountLinks.forEach(account => {
      const link = screen.getByRole('link', {
        name: new RegExp(account.name, 'i'),
      });
      expect(link).toHaveAttribute('href', account.url);
    });
  });

  test('should have proper accessibility attributes', () => {
    render(<Account />);

    const nav = screen.getByRole('navigation');
    expect(nav).toHaveAttribute('aria-label', 'account links');
  });
});
