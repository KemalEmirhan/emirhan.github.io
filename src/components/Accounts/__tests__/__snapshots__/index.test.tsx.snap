// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Account Component > should render 1`] = `
<div>
  <nav
    aria-label="account links"
    class="border-gray-200 max-w-4xl mx-auto px-4 sm:px-6 py-6 sm:py-8"
  >
    <div
      class="flex flex-wrap justify-center gap-4 sm:gap-6"
    >
      <a
        class="flex items-center gap-2 sm:gap-3 px-3 sm:px-4 py-2 sm:py-3 rounded-lg bg-white border border-gray-200 hover:border-gray-300 hover:shadow-md transition-all duration-150 group"
        href="https://github.com/KemalEmirhan"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-github text-gray-700 group-hover:text-gray-900 transition-colors duration-150"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
          />
          <path
            d="M9 18c-4.51 2-5-2-7-2"
          />
        </svg>
        <span
          class="text-sm sm:text-base text-gray-700 font-medium group-hover:text-gray-900 transition-colors duration-150"
        >
          Github
        </span>
      </a>
      <a
        class="flex items-center gap-2 sm:gap-3 px-3 sm:px-4 py-2 sm:py-3 rounded-lg bg-white border border-gray-200 hover:border-gray-300 hover:shadow-md transition-all duration-150 group"
        href="https://www.linkedin.com/in/emirhan-kemal-kosem-259569b3/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-linkedin text-gray-700 group-hover:text-gray-900 transition-colors duration-150"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
          />
          <rect
            height="12"
            width="4"
            x="2"
            y="9"
          />
          <circle
            cx="4"
            cy="4"
            r="2"
          />
        </svg>
        <span
          class="text-sm sm:text-base text-gray-700 font-medium group-hover:text-gray-900 transition-colors duration-150"
        >
          Linkedin
        </span>
      </a>
      <a
        class="flex items-center gap-2 sm:gap-3 px-3 sm:px-4 py-2 sm:py-3 rounded-lg bg-white border border-gray-200 hover:border-gray-300 hover:shadow-md transition-all duration-150 group"
        href="https://medium.com/@emirhankemalkosem"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-book-open text-gray-700 group-hover:text-gray-900 transition-colors duration-150"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 7v14"
          />
          <path
            d="M3 18a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h5a4 4 0 0 1 4 4 4 4 0 0 1 4-4h5a1 1 0 0 1 1 1v13a1 1 0 0 1-1 1h-6a3 3 0 0 0-3 3 3 3 0 0 0-3-3z"
          />
        </svg>
        <span
          class="text-sm sm:text-base text-gray-700 font-medium group-hover:text-gray-900 transition-colors duration-150"
        >
          Medium
        </span>
      </a>
      <a
        class="flex items-center gap-2 sm:gap-3 px-3 sm:px-4 py-2 sm:py-3 rounded-lg bg-white border border-gray-200 hover:border-gray-300 hover:shadow-md transition-all duration-150 group"
        href="https://twitter.com/KsemEmir"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-twitter text-gray-700 group-hover:text-gray-900 transition-colors duration-150"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"
          />
        </svg>
        <span
          class="text-sm sm:text-base text-gray-700 font-medium group-hover:text-gray-900 transition-colors duration-150"
        >
          Twitter
        </span>
      </a>
      <a
        class="flex items-center gap-2 sm:gap-3 px-3 sm:px-4 py-2 sm:py-3 rounded-lg bg-white border border-gray-200 hover:border-gray-300 hover:shadow-md transition-all duration-150 group"
        href="https://www.instagram.com/emirhankemalkosem/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-instagram text-gray-700 group-hover:text-gray-900 transition-colors duration-150"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="20"
            rx="5"
            ry="5"
            width="20"
            x="2"
            y="2"
          />
          <path
            d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
          />
          <line
            x1="17.5"
            x2="17.51"
            y1="6.5"
            y2="6.5"
          />
        </svg>
        <span
          class="text-sm sm:text-base text-gray-700 font-medium group-hover:text-gray-900 transition-colors duration-150"
        >
          Instagram
        </span>
      </a>
    </div>
  </nav>
</div>
`;
